File: Backend/SmartParkingSystem/src/main/java/com/example/SmartParkingSystem/entities/Reservation.java
package com.example.SmartParkingSystem.entities;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Reservation {
    private Long id;
    private Integer driverId;
    private Integer spotId;
    private ReservationStatus status;
    private LocalDateTime checkIn;
    private LocalDateTime checkOut;
    private LocalDateTime scheduledCheckIn;
    private LocalDateTime scheduledCheckOut;
    private BigDecimal amount;
    private String paymentMethod;
    private String transactionId;
    private LocalDateTime createdAt;
}

File: Frontend/src/types/Reservation.ts
export enum ReservationStatus {
  PENDING = "PENDING",
  ACTIVE = "ACTIVE",
  COMPLETED = "COMPLETED",
  NO_SHOW = "NO_SHOW",
}

export interface Reservation {
  id: string;
  location: string;
  name: string;
  spotNumber: number;
  status: ReservationStatus;
  checkIn: string | null;
  checkOut: string | null;
  scheduledCheckIn: string;
  scheduledCheckOut: string;
  amount: string;
  paymentMethod: string;
  transactionId: string;
  createdAt: string;
}
